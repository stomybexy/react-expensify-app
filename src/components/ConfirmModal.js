import React from 'react';
import Modal from 'react-modal';

const ConfirmModal = (props) => (
    <Modal
        isOpen={props.isOpen}
        onRequestClose={props.onCancel}
        contentLabel={props.title}
        closeTimeoutMS={200}
        className="modal"
        ariaHideApp={false}
    >
        <p className="modal__body">{props.message || 'Do you confirm ?'}</p>

        <div className="modal__actions">
            <button id="no" className="button" onClick={props.onCancel}>
                No
            </button>
            <button id="yes" className="button  button--danger" onClick={props.onConfirm}>
                Yes
            </button>
        </div>
    </Modal>
);

export default ConfirmModal;