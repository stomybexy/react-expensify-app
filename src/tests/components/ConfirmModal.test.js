import React from 'react';
import { shallow } from 'enzyme';
import ConfirmModal from '../../components/ConfirmModal';

let onCancel, onConfirm, wrapper;

beforeEach(() => {
    onCancel = jest.fn();
    onConfirm = jest.fn();
    wrapper = shallow(
        <ConfirmModal
            onConfirm={onConfirm}
            onCancel={onCancel}
            title="Modal"
            message="Confirm your action ?"
            isOpen={true}
        />
    )
});

test('It should render ConfirmModal correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('It should cancel confirm on no button click', () => {
    wrapper.find('#no').simulate('click');
    expect(onCancel).toHaveBeenCalled();
});
test('It should cancel confirm on modal dismiss', () => {
    wrapper.find('Modal').props().onRequestClose();
    expect(onCancel).toHaveBeenCalled();
});
test('It should confirm on yes button click', () => {
    wrapper.find('#yes').simulate('click');
    expect(onConfirm).toHaveBeenCalled();
});
