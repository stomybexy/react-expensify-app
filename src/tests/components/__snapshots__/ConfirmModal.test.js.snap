// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render ConfirmModal correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Modal"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <p
    className="modal__body"
  >
    Confirm your action ?
  </p>
  <div
    className="modal__actions"
  >
    <button
      className="button"
      id="no"
      onClick={[Function]}
    >
      No
    </button>
    <button
      className="button  button--danger"
      id="yes"
      onClick={[Function]}
    >
      Yes
    </button>
  </div>
</Modal>
`;
